### ISSUE TITLE
.netstandard2.0 support?

### ISSUE THREAD

**Issue by ikijano** [amantinband/throw#64]
Looks a fantastic library and I like to use it in my work project but unfortunate I'm stuck in .net framework world so i'm unable to use this library :(
Have you ever considered supporting .netstandard2.0?

I played a little bit library source code and I managed get it compile when targeting to netstandard2.0 with langversion 10 using help of [IsExternalInit](https://www.nuget.org/packages/IsExternalInit) and [Nullable](https://www.nuget.org/packages/Nullable) packages and re-ordering of namespace and using statements on couple files because of stylecop.
It looks like there is no actual show stoppers in library code only some tests needs slight modifications. When running  tests "only" 206/1994 tests fails on .net framework 4.8. Common pattern seems to be that ```ArgumentException``` message format is slightly different on .net framework vs .net core. Tests are assuming blindly .net core message format instead of using real message format implemented by runtime which i considered to be implementation bug. If runtime implementation ever changes all these test will fail.

In example following test will fail on .net framework 4.8:

```c#
    public void Throw_WhenCustomExceptionMessage_ShouldThrowArgumentExceptionWithCustomMessage()
    {
        // Arrange
        ExceptionCustomizations exceptionCustomizations = ParameterConstants.CustomMessage;

        // Act
        Action action = () => ExceptionThrower.Throw(ParameterConstants.ParamName, exceptionCustomizations);

        // Assert
        action.Should().ThrowExactly<ArgumentException>().WithMessage($"{ParameterConstants.CustomMessage} (Parameter '{ParameterConstants.ParamName}')");
    }
```

Message:
```
Expected exception message to match the equivalent of "custom message (Parameter 'paramName')", but "custom message
Parameter name: paramName" does not.
```

if I change test to following form (not assuming any message format implementation details) test will succeed:

```c#
        // Assert
        action.Should().ThrowExactly<ArgumentException>().WithMessage(new ArgumentException($"{ParameterConstants.CustomMessage}", ParameterConstants.ParamName).Message);
```

test repo (no test modifications)
https://github.com/ikijano/throw/tree/wip-netstandard2

**Comment by RobThree** [comment]
I agree - a library should support Netstandard 2.0 whenever possible. Also agree on relying on string literals isn't the way go, but on the other hand the suggested 'fix' doesn't actually test anything either; it just mirrors internal behavior.

**Comment by ikijano** [comment]
I 'fixed' most of test in my test repo. So far everything is good.

I think, I could live with my fork until official library supports netstandard2.0 or i can move on to .net6+ era.

Of course it would be nice if official library would use more predicted message checks in tests so back porting new features to unofficial lib would be simpler.

**Comment by RobThree** [comment]
@ikijano Why don't you do a PR?

**Comment by ikijano** [comment]
I will do when I finish test fixes and do some cleanups.

**Comment by bstordrup** [comment]
Completely agree. IMO, supporting netstandard2.0 in a library project is essential.

**Comment by ikijano** [comment]
@RobThree Any suggestion how to modify test properly?
I would like to step into this again and took another take.

**Comment by RobThree** [comment]
No, sorry, After the initial "hmm, this looks cool" I've never gotten around to actually using it and now, few years later, I'm not sure I still would be interested.
